controllers/mutation_reconcile_looper.go

```diff
@@ -801,19 +804,47 @@ func (m *MutationReconcileLooper) getValues(ctx context.Context, obj *v1alpha1.M
 			return nil, fmt.Errorf("could not read values file: %w", err)
 		}
 
-		var data map[string]any
 		if err := yaml.Unmarshal(dataFile, &data); err != nil {
 			return nil, fmt.Errorf("failed to unmarshal values: %w", err)
 		}
-
 		var found bool
 		if obj.ValuesFrom.FluxSource.SubPath != "" {
 			data, found = extractSubpath(data, obj.ValuesFrom.FluxSource.SubPath)
 			if !found {
 				return nil, errors.New("subPath not found")
 			}
 		}
+	} else if obj.ValuesFrom.ConfigMapSource != nil {
```

Comment:
Now I wonder if these could be defined together? Some values come from Config, others from FluxSource?

Reply:
On a side-note. Can you please extract these into functions? This if/else block is now rather huge and hard to follow.